arroyo-controller/src/job_controller/checkpointer.rs

```diff
@@ -161,9 +219,16 @@ impl CheckpointState {
                     grpc::TaskCheckpointEventType::FinishedSync => {
                         api::TaskCheckpointEventType::CheckpointSyncFinished
                     }
+                    grpc::TaskCheckpointEventType::FinishedCommit => {
+                        api::TaskCheckpointEventType::CheckpointPreCommit
+                    }
                 } as i32,
             });
 
+        if grpc::TaskCheckpointEventType::FinishedCommit == c.event_type() {
+            panic!("shouldn't receive finished commits while checkpointing");
```

Comment:
panics in the job controller should only be used for cases that indicate an unrecoverable programming error

Reply:
I'll have it bail, as this likely means that it won't be able to checkpoint.